hotham-asset-client/src/client.rs

```diff
@@ -0,0 +1,218 @@
+use crate::message::Message;
+use crate::AssetUpdatedMessage;
+use anyhow::{bail, Context, Result};
+use futures_util::{StreamExt, TryStreamExt};
+use quinn::{ClientConfig, Endpoint};
+use std::sync::Arc;
+use tokio::sync::mpsc::Sender;
+
+const BUFFER_SIZE: usize = 104_857_600; // 100MB
+
+pub async fn watch(asset_names: Vec<String>, sender: Sender<AssetUpdatedMessage>) -> Result<()> {
+    run_client(asset_names, sender).await
+}
+
+async fn run_client(
+    mut asset_names: Vec<String>,
+    sender: Sender<AssetUpdatedMessage>,
+) -> Result<()> {
+    let server_addr = "192.168.1.124:5000".parse().unwrap();
```

Comment:
This hardcoded address is problematic. Can we use a parameter or an environment variable instead?

Reply:
Yeah I went back and forth on a couple of approaches for this. We're missing a good general solution for configuration settings like these!

For now I've stuck to `option_env` which should check on the host machine for an environment variable. I made this optional as it would be extremely annoying to make this mandatory, and the error message will bubble back up to the engine if anybody is looking. I also made it a *host* variable as setting environment variables on Android is possible, but annoying (again, it would be good to have a good general solution for this).